torch/_subclasses/meta_utils.py

```diff
@@ -225,6 +225,41 @@ def describe_tensor(self, t: torch.Tensor, recurse: bool = True):
             # sym_sizes_strides_storage_offset (empty_create)
             stride = t.stride()
 
+        # NB: this technically should refer to functorch unwrapped tensor, but
+        # I am (perhaps abusively) using it to store both the functorch and
+        # non-functorch functional tensor
+        unwrapped = None
+        autograd_meta_from = None
+        current_level = None
+        if is_batchedtensor_v or is_gradtrackingtensor_v:
+            unwrapped = self.describe_tensor(get_unwrapped(t))
```

Comment:
Is this recursive? get_unwrapped only removes one layer of functorch wrappers

Reply:
Describe tensor is the method here, so yes, we will recursively build descs for each level of unwrapping until we bottom out

Reply:
Answer: yes